helix-core/src/main/java/org/apache/helix/controller/stages/TopStateHandoffReportStage.java

```diff
@@ -471,6 +474,7 @@ private void reportTopStateComesBack(ResourceControllerDataProvider cache, Map<S
       }
     }
 
+    // TODO: why are we checking if handoff happened within threshold here? it should be reported either way.
```

Comment:
@junkaixue i have a question here. Why are we updating topstate handoff duration only if it's within threshold. This means that if threshold is set to "x" ms and any partition's top state has recovered beyond "x" ms then we would not report it. Basically same behaviour as reportSingleTopStateHandoff. (If my understanding is correct then i can remove this, i have also changed test to verify this.)

Reply:
Filed #2383. Let's move this discussion there.